### PR TITLE
doc: pull supported boards from sample.yaml

### PR DESCRIPTION
**Pull request by gautesl**
Modified the `table-from-rows` directive to no longer support
the `sample-yaml-rows` option. Instead, a new directive
`table-from-sample-yaml` has been added which reads supported
boards from the integration_platforms section of the
`sample.yaml` file.

All samples and most applications have had their requirements
section updated to use the new directive.

A lot of the requirement section tables are changing as a
consequence of this modification. A complete list of all the
changes can be found in `changes.txt`. Files in the nrf folder
that still use the old extension can be found in
`manually_maintained.txt`.

[changes.txt](https://github.com/nrfconnect/sdk-nrf/files/8441051/changes.txt)
[manually_maintained.txt](https://github.com/nrfconnect/sdk-nrf/files/8441057/manually_maintained.txt.txt)

### CODE REVIEW

File: samples/tfm/tfm_hello_world/README.rst
Comment (joerchan) [inline samples/tfm/tfm_hello_world/README.rst:17]: Can I do this change immediately to https://github.com/nrfconnect/sdk-nrf/pull/7148/files#diff-e40ce80d37ad512bc8ce14a370eb1fa2177b03d70e9d0903b1815174342974afR17
or do I need to wait for this PR to be merged?

**Review by joerchan — APPROVED** [review]

**Comment by PerMac** [comment]
Please add any dummy change (e.g. extra white space in the commit msg) to the PR, so the SHA is changed. The issue with OSS history check was fixed, but it requires change in the SHA to reload the fixed workflow.

**Comment by frkv** [comment]
There is an issue in the tables for crypto samples with regards to devices that has TrustZone (has both NS and secure only builds)

In the existing documentation (latest official release) there are multiple lines for NS/S build targets for nRF5340 and nRF9160, but inthis PR  there is a combination of NS/S targets for nRF5340 (only), while nRF9160 still appears as split

Please see [this link](http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-7308/nrf/samples/crypto/aes_cbc/README.html)

Unsure if the root-cause is this specific PR, but this is the first location where I've seen this inconsistency

**Comment by joerchan** [comment]
> Please add any dummy change (e.g. extra white space in the commit msg) to the PR, so the SHA is changed. The issue with OSS history check was fixed, but it requires change in the SHA to reload the fixed workflow.

`git commit --amend --no-edit` works well for this.

**Comment by gautesl** [comment]
> 

The reason for this is that the [sample_board_rows.txt file](https://github.com/nrfconnect/sdk-nrf/blob/7fcb47b90c5d3de729aa3cc0a8b71ee7f9744c60/doc/nrf/includes/sample_board_rows.txt) has a rule for `nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns`, but not one combining `nrf9160dk_nrf9160` and `nrf9160dk_nrf9160_ns`.

Should this be added? Are there any other pairs of boards that should be grouped together? Currently the only combinatory table entries are the following:
- `nrf5340dk_nrf5340_cpuapp_and_cpuapp_ns`
- `nrf5340dk_nrf5340_cpuapp_and_cpunet`
- `thingy53_nrf5340_cpuapp_and_cpuapp_ns`

@carlescufi did you have an opinion on this?

**Comment by joerchan** [comment]
What does the grouping give us?

Seems that we are adding additional maintenance with that, and it is not complete.
Why isn't thingy53_nrf5340_cpuapp_and_cpunet a grouping.
We would also be missing thingy91.

**Review by simensrostad — APPROVED** [review]

**Comment by umapraseeda** [comment]
> There is an issue in the tables for crypto samples with regards to devices that has TrustZone (has both NS and secure only builds)
> 
> In the existing documentation (latest official release) there are multiple lines for NS/S build targets for nRF5340 and nRF9160, but in this PR there is a combination of NS/S targets for nRF5340 (only), while nRF9160 still appears as split
> 
> Please see [this link](http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-7308/nrf/samples/crypto/aes_cbc/README.html)
> 
> Unsure if the root-cause is this specific PR, but this is the first location where I've seen this inconsistency
> 
> _My personal preference for this fix is to standardized on the methodology of listing DKs (and PCA version) **in one line** and have multiple build-targets in a cell._
> 
> Maybe @umapraseeda can chime in on this...

I think having a single row for DKs and listing the build targets associated with that DK in the same row as in the following image would look nice:

image

And if we are doing this, then it has to be consistent for all the DKs, wherever applicable. @carlescufi, @b-gent, your thoughts?

**Review by shantha-14 — APPROVED** [review]

**Review by VeijoPesonen — APPROVED** [review]
LwM2M Client sample looks ok.

**Review by tomchy — APPROVED** [review]

**Review by mia-ko — COMMENTED** [review]
Looks ok for the Bluetooth mesh samples

**Review by mia-ko — APPROVED** [review]

**Review by junqingzou — APPROVED** [review]
Updates for Serial LTE Modem all fine

**Comment by gautesl** [comment]
> > There is an issue in the tables for crypto samples with regards to devices that has TrustZone (has both NS and secure only builds)
> > In the existing documentation (latest official release) there are multiple lines for NS/S build targets for nRF5340 and nRF9160, but in this PR there is a combination of NS/S targets for nRF5340 (only), while nRF9160 still appears as split
> > Please see [this link](http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-7308/nrf/samples/crypto/aes_cbc/README.html)
> > Unsure if the root-cause is this specific PR, but this is the first location where I've seen this inconsistency
> > _My personal preference for this fix is to standardized on the methodology of listing DKs (and PCA version) **in one line** and have multiple build-targets in a cell._
> > Maybe @umapraseeda can chime in on this...
> 
> I think having a single row for DKs and listing the build targets associated with that DK in the same row as in the following image would look nice:
> 
> image
> 
> And if we are doing this, then it has to be consistent for all the DKs, wherever applicable. @carlescufi, @b-gent, your thoughts?

After a discussion on this, merged rows continue to follow the hard-coded options listed in `sample_board_rows.txt`. An entry merging `nrf9160dk_nrf9160` and `nrf9160dk_nrf9160_ns` has been added to this file.

**Review by frkv — APPROVED** [review]
LGTM (Crypto)

**Comment by mbolivar-nordic** [comment]
@gautesl the test failures look unrelated to me. Can you confirm? If so, I can force merge.

**Comment by gautesl** [comment]
> @gautesl the test failures look unrelated to me. Can you confirm? If so, I can force merge.

@mbolivar-nordic They look unrelated to me too

**Comment by PerMac** [comment]
@mbolivar-nordic @gautesl  It looks like the issues are with BLE downstream job. Failure seems not related to this PR

**Comment by thst-nordic** [comment]
only rst files being touched, so I think no integration testing is needed.  
I'll talk with testers to see what we need to adjust in test-spec. 
image

**Comment by gautesl** [comment]
Introduced the change to `samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst` and `samples/caf_sensor_manager/README.rst` to keep the requirements sections consistent.

**Review by umapraseeda — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]
LGTM (/samples/nrf9160)

**Comment by jayteemo** [comment]
sorry i missed this PR, a new sample was just merged which needs this change:
https://github.com/nrfconnect/sdk-nrf/blob/main/samples/nrf9160/nrf_cloud_rest_cell_pos/README.rst

**Review by plskeggs — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
@jayteemo are you asking for a change?

**Review by jayteemo — CHANGES_REQUESTED** [review]
sorry i missed this PR, a new sample was just merged which needs this change:
https://github.com/nrfconnect/sdk-nrf/blob/main/samples/nrf9160/nrf_cloud_rest_cell_pos/README.rst

**Comment by jayteemo** [comment]
> @jayteemo are you asking for a change?

an addition... so yes. 
updated.

**Comment by gautesl** [comment]
Introduced the change to `samples/nrf9160/nrf_cloud_rest_cell_pos/README.rst`.

**Review by jayteemo — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
Needs rebase

File: doc/_extensions/table_from_rows.py
Comment (gmarull) [inline doc/_extensions/table_from_rows.py:139]: note: to make this more future proof, you could look for `{x}` + `{x}_ns` or `{x}_cpuapp` + `{x}_cpunet` (maybe using regexes). it's not nice to have a dependency on boards at an extension level.

File: doc/_extensions/table_from_rows.py
Comment (gmarull) [inline doc/_extensions/table_from_rows.py:158]: this is hard to read/follow, maybe (not tesed_:

```python
rel_path = os.path.relpath(path, self.env.srcdir)
dir_path = Path(self.config.table_from_rows_base_dir).parent / rel_path
```

File: doc/_extensions/table_from_rows.py
Comment (gmarull) [inline doc/_extensions/table_from_rows.py:181]: note: these may be obviated, they won't be catched and produce an error anyway. YAMLError may potentially include syntax error information in the traceback (hidden here).

**Review by gmarull — COMMENTED** [review]
please check my comments

**Comment by gautesl** [comment]
Made groupings dynamic, rows will now group with other rows that are equal in all columns except for the build target when using ``table_from_sample_yaml``. Also removed the need to specify the same include-file and header every time the directive is used. It can now be used like this:
```rst
.. table_from_sample_yaml::
```

File: doc/_extensions/table_from_rows.py
Comment (gmarull) [inline doc/_extensions/table_from_rows.py:135]: nit: should be deleted

File: doc/_extensions/table_from_rows.py
Comment (gautesl) [reply doc/_extensions/table_from_rows.py]: This piece of code is taking care of an edge case where the supplied path did not point to the build directory. It has been restructured and added comments to make it easier to follow.

**Review by gmarull — APPROVED** [review]
great job

**Pull request merged**
